src/pages/RFOX/components/Unstake/hooks/useRfoxUnstake.tsx

```diff
@@ -0,0 +1,219 @@
+import type { AccountId } from '@shapeshiftoss/caip'
+import { type AssetId, fromAccountId, fromAssetId } from '@shapeshiftoss/caip'
+import { CONTRACT_INTERACTION } from '@shapeshiftoss/chain-adapters'
+import type { UseMutationResult } from '@tanstack/react-query'
+import { useMutation, type UseQueryResult } from '@tanstack/react-query'
+import { foxStakingV1Abi } from 'contracts/abis/FoxStakingV1'
+import { RFOX_PROXY_CONTRACT_ADDRESS } from 'contracts/constants'
+import { useMemo } from 'react'
+import type { UseFormReturn } from 'react-hook-form'
+import { encodeFunctionData } from 'viem'
+import type { UseQueryReturnType } from 'wagmi/query'
+import { useEvmFees } from 'hooks/queries/useEvmFees'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { fromBaseUnit, toBaseUnit } from 'lib/math'
+import type { EvmFees } from 'lib/utils/evm'
+import {
+  assertGetEvmChainAdapter,
+  buildAndBroadcast,
+  createBuildCustomTxInput,
+} from 'lib/utils/evm'
+import { useStakingBalanceOfQuery } from 'pages/RFOX/hooks/useStakingBalanceOfQuery'
+import { useStakingInfoQuery } from 'pages/RFOX/hooks/useStakingInfoQuery'
+import {
+  selectAccountNumberByAccountId,
+  selectAssetById,
+  selectFeeAssetByChainId,
+  selectTxById,
+} from 'state/slices/selectors'
+import { serializeTxIndex } from 'state/slices/txHistorySlice/utils'
+import { useAppSelector } from 'state/store'
+
+import type { UnstakeInputValues } from '../types'
+
+type UseRfoxUnstakeProps = {
+  stakingAssetId: AssetId
+  stakingAssetAccountId: AccountId | undefined
+  amountCryptoBaseUnit: string
+  methods: UseFormReturn<UnstakeInputValues> | undefined
+  setUnstakeTxid: ((txId: string) => void) | undefined
+  unstakeTxid: string | undefined
+}
+type UseRfoxUnstake = (props: UseRfoxUnstakeProps) => {
+  unstakeFeesQuery: UseQueryResult<EvmFees, Error>
+  unstakeMutation: UseMutationResult<string | undefined, Error, void, unknown>
+  isUnstakeTxPending: boolean
+  isGetUnstakeFeesEnabled: boolean
+  userStakingBalanceOfQuery: UseQueryReturnType<string, unknown>
+  newContractBalanceOfQuery: UseQueryReturnType<string, unknown>
+  newShareOfPoolPercentage: string
+}
+
+export const useRfoxUnstake: UseRfoxUnstake = ({
+  methods,
+  stakingAssetId,
+  stakingAssetAccountId,
+  amountCryptoBaseUnit,
+  setUnstakeTxid,
+  unstakeTxid,
+}) => {
```

Comment:
Non-blocking, personally not a fan of a whole type dedicated to a function - 2 layers of indirection just to check the types of the params when we could do params/return types and inline them like this:

```suggestion
type UseRfoxUnstakeProps = {
  stakingAssetId: AssetId
  stakingAssetAccountId: AccountId | undefined
  amountCryptoBaseUnit: string
  methods: UseFormReturn<UnstakeInputValues> | undefined
  setUnstakeTxid: ((txId: string) => void) | undefined
  unstakeTxid: string | undefined
}

type UseRfoxUnstakeResult = {
  unstakeFeesQuery: UseQueryResult<EvmFees, Error>
  unstakeMutation: UseMutationResult<string | undefined, Error, void, unknown>
  isUnstakeTxPending: boolean
  isGetUnstakeFeesEnabled: boolean
  userStakingBalanceOfQuery: UseQueryReturnType<string, unknown>
  newContractBalanceOfQuery: UseQueryReturnType<string, unknown>
  newShareOfPoolPercentage: string
}

export const useRfoxUnstake: UseRfoxUnstake = ({
  methods,
  stakingAssetId,
  stakingAssetAccountId,
  amountCryptoBaseUnit,
  setUnstakeTxid,
  unstakeTxid,
}: UseRfoxUnstakeProps): UseRfoxUnstakeResult => {
```

Reply:
but even then its 1 layer of indirection we dont need, i much prefer inline types unless we need to use them externally (which we dont seem to?)

```suggestion
export const useRfoxUnstake: UseRfoxUnstake = ({
  methods,
  stakingAssetId,
  stakingAssetAccountId,
  amountCryptoBaseUnit,
  setUnstakeTxid,
  unstakeTxid,
}: {
  stakingAssetId: AssetId
  stakingAssetAccountId: AccountId | undefined
  amountCryptoBaseUnit: string
  methods: UseFormReturn<UnstakeInputValues> | undefined
  setUnstakeTxid: ((txId: string) => void) | undefined
  unstakeTxid: string | undefined
}) => {
```

Reply:
Personally against inlining types (especially as they get quite big) for readbiility - colocated instead of inline always gives some space to the component to breathe (meaning you don't have to visually parse this whole blob and it's above) but agree the whole function dedicated to this hook is probably too much!

Reply:
b9028dfb77